src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -3562,7 +3635,7 @@ export function generateSamplePointsCube(
       mipLevel,
       arrayIndex: args.arrayIndex ? makeRangeValue(args.arrayIndex, i, 2) : undefined,
       bias,
-      depthRef: args.depthRef ? makeRangeValue({ num: 1, type: 'f32' }, i, 5) : undefined,
+      depthRef: args.depthRef ? makeRandValue({ num: 3, type: 'u32' }, i, 5) / 2 : undefined,
```

Comment:
Q: I'm unsure about this line: we have random texel values between [0, 1]. is it we want depthRef being some random number of [0, 1.5]?

Why is type 'u32' but not 'f32' for `textureSampleCompare` and other returns f32?

Reply:
We use `u32` so `makeRandValue` returns an integer `[0, (num -1]]` (so 0, 1, 2) then dividing by 2 gives 0, 0.5, 1. Added a comment as to why